joomla/joomla-cms
# Update to FormField.php - Code roboration / Bugfix ( 1 minute fix )

Proper checking of isset conditions, roborating the code. Otherwise, in case there is no form set, code will crash ( with error : "Call to a member function getXml() on null" ). Please Update the code accordingly.

Pull Request for Issue # .

### Summary of Changes

Adds isset check to prevent crash of code, when improperly assumed condition ( ->form exists ) is not met. Roborates the Code.

### Testing Instructions


```

use Joomla\CMS\Form\Field\TextField;

$feld = new TextField();
$xmlstring = '
<field
		name="mytest"
		label="My Test Standalone Field"
		type="test" 
/> 
';
$xml = new SimpleXMLElement( $xmlstring );
$feld->setup( $xml );
$field =  $feld->renderField();

```
As you will see, the Application crashes, simply because of the afore mentioned inpropper check, that falsely and blindly assumes, that there is a contextual form assosiated with the field ( which is never the case, when you set up a 'standalone' field ).
The source of the crash can be found here :

```
/libraries/src/Form/FormField.php line 1018+ :

        $options['inlineHelp'] = isset($this->form->getXml()->config->inlinehelp['button'])
            ? ((string) $this->form->getXml()->config->inlinehelp['button'] == 'show' ?: false)
            : false;

```

Again, the inaccuracy of the code here does not even have to to with Joomla per se, it's simply good coding practice to NEVER blindly assume anything at all, but to make sure. But of course, in this specific case, not fixing it would mean the completely unneccesary abandonment of a whole feature in Joomla! : The ability to have Joomla! FormFields set up in a 'standalone' manner, which are a beautiful feature, since they profit from all the beautiful typical Joomla perks in regards to FormField's features, like e.g. skinnability ( layouts ), etc.


### Actual result BEFORE applying this Pull Request

application crash with error : "Call to a member function getXml() on null" 

### Expected result AFTER applying this Pull Request

just works even when there is no form present / set, aka when you set up a 'standalone' Field as exemplified in the above code.

### Link to documentations
Please select:
